Added CVE-2021-3584 (#1845)
theforeman/theforeman.org: security.md

### Disclosure details

#### <a id="2021-3584"></a>CVE-2021-3584: Remote code execution through Sendmail configuration

> Sendmail location and arguments, available via Administer - Settings, both accept arbitrary strings and pass them into shell. By default, only Foreman super administrator can access settings.

Reviewer: ```suggestion
Sendmail location and arguments, available via Administer - Settings, both accept arbitrary strings and pass them into shell. By default, only Foreman super administrators can access settings.
```

Reviewer: Apologies for not catching this the first time
```suggestion
Sendmail location and arguments, available via Administer - Settings, both accept arbitrary strings and pass them into shell. By default, only Foreman super administrators can access settings. However, non-admin users can be granted the permission to edit settings.
```

Reviewer 2: This is incorrect. By default users with the `Manager` or `System admin` roles also have permission to edit settings.

Author: Corrected.

> Sendmail location and arguments, available via Administer - Settings, both accept arbitrary strings and pass them into shell. By default, only Foreman super administrators can access settings.

Reviewer 2: ```suggestion
Sendmail location and arguments, available via Administer - Settings, both accept arbitrary strings and pass them into shell. By default, Foreman administrators as well as "Manager" and "System Administrator" roles can modify settings.
```

> *Mitigation:* Verify the both settings and remove edit_settings permissions to all roles and users until fixed. Alternatively, create settings named sendmail_location and sendmail_arguments in `settings.yaml` file to override the UI and make the values read-only.

Reviewer: ```suggestion
*Mitigation:* Verify the both settings and remove edit_settings permissions from all roles and users until fixed. Alternatively, create settings named `sendmail_location` and `sendmail_arguments` in `settings.yaml` file to override the UI and make the values read-only. Note that the installer will remove these again.
```

> * Affects Foreman veersion 1.15 and higher

Reviewer: ```suggestion
* Affects Foreman version 1.15 and higher
```

Author: Looks quite Dutch :-) veersion 🇳🇱

Reviewer: The Dutch translation is `versie`. `veer` means feather or spring (the metal kind that you can compress and it'll take its old shape again) but there's no `veersion` ;)

Reviewer: Typo is still here

Author: Damn, rebased.

> *Mitigation:* Verify the both settings and remove edit_settings permissions from all roles and users until fixed. Alternatively, create settings named `sendmail_location` and `sendmail_arguments` in `settings.yaml` file to override the UI and make the values read-only. Note that the installer will remove these again.

Reviewer: Just wondering: will users understand that they really should use `:sendmail_location`? Perhaps an example section is good?

Reviewer: I still think this is better. Otherwise users may create an entry that ends up being unused. A literal example is even better.

Author: Done.

> *Mitigation:* Verify the both settings and remove edit_settings permissions from all roles and users until fixed. Alternatively, create settings named `sendmail_location` and `sendmail_arguments` in `settings.yaml` file to override the UI and make the values read-only. Note that the installer will remove these again.

Reviewer 2: ```suggestion
*Mitigation:* Verify the value of both settings and remove the `edit_settings` permissions from any untrusted users until fixed. Alternatively, create settings named `sendmail_location` and `sendmail_arguments` in `settings.yaml` file to override the UI and make the values read-only. Note that the installer will remove these again.
```

> * Fix released in Foreman 2.4.1, 2.5.1 and higher

Author: To be confirmed by @upadhyeammit

Author: Confirmed by Amit thanks.

### Disclosure details

#### <a id="2021-3584"></a>CVE-2021-3584: Remote code execution through Sendmail configuration

Sendmail location and arguments, available via Administer - Settings, both accept arbitrary strings and pass them into shell. By default, only Foreman super administrator can access settings.

> *Mitigation:* Verify the both settings and remove edit_settings permissions to all roles and users until fixed. Alternatively, create entries in `settings.yaml` to override UI settings and make the read-only:
> 
>     :sendmail_location: "/usr/sbin/sendmail"
>     :sendmail_arguments: "-i"

Reviewer: ````suggestion
*Mitigation 1:* Verify the both settings and remove edit_settings permissions to all roles and users until fixed.

*Mitigation 2:* Create entries in `settings.yaml` to override UI settings and make them read-only. Note that the installer will revert these changes. [#32827](https://projects.theforeman.org/issues/32827) will add options to the installer which makes it supported.

```yaml
:sendmail_location: "/usr/sbin/sendmail"
:sendmail_arguments: "-i"
```
````
